### PR TITLE
Adding a minimal text example

### PR DESCRIPTION
**Pull request by AdLucem**
This is a PR for a small text example- training the RNN to predict the last letter of the word "hello".

The point of this example is:
- checking if the RNNs can overfit on one example (it does)
- putting a `representation` function- currently a simple one-hot mapping for letters- that I can extend or elaborate into a representation layer later
- putting a `prediction` function that takes the result of the RNN and maps it to the required label (in this case, one of four letters)

### CODE REVIEW

**Comment by junjihashimoto** [comment]
LGTM!

File: examples/rnn/LSTM.hs [hunk 1]
```diff
@@ -60,24 +60,16 @@ instance RecurrentCell LSTMCell where
...
-      let fg = [fg_ih, fg_hh, fg_b]
-      let og = [og_ih, og_hh, og_b]
-      let hg = [hg_ih, hg_hh, hg_b]
+      w_ih <- makeIndependent =<< randn' [inf, hf]
+      w_hh <- makeIndependent =<< randn' [hf, hf]
+      b <- makeIndependent =<< randn' [1, hf]
+      let ig = [w_ih, w_hh, b]
+      let fg = [w_ih, w_hh, b]
+      let og = [w_ih, w_hh, b]
+      let hg = [w_ih, w_hh, b]
```
Comment (junjihashimoto) [inline examples/rnn/LSTM.hs]: These codes works at first step only.
In training loop, we call makeIndependent for each parameters.
https://github.com/hasktorch/hasktorch/blob/master/examples/elman/Main.hs#L48
So if you want to merge these variables, you should remove duplicated variable of data-LSTM or Paramertrized's one.

Comment (austinvhuang) [reply examples/rnn/LSTM.hs]: @AdLucem if you can fix this and revert the seq2seq commit so that's in a different feature branch I can merge this PR.

Comment (AdLucem) [reply examples/rnn/LSTM.hs]: Fixed!

**Comment by junjihashimoto** [comment]
I do not understand the cause of [CI's error](https://circleci.com/gh/hasktorch/hasktorch/1779 ).
The error is as follows.
It returns "ExitFailure (-9)".

```
Cuda
  CudaSpec
                 
Progress 1/2: ffi                 ffi> Test suite spec failed
Progress 1/2: ffi                 Completed 2 action(s).
Test suite failure for package ffi-0.1.0.0
    spec:  exited with: ExitFailure (-9)
Logs printed to console

Exited with code 1
```

If the memory is not enough, the output is as follows, and it returns "ExitFailure 1".

```
$ ulimit -v 1048576
$ stack test libtorch-ffi --ta "-m Cuda"
Cabal file warning in/home/junji-hashimoto/git/hasktorch/libtorch-ffi/libtorch-ffi.cabal@107:3: The field "cxx-options" is available since Cabal [2,1]
Cabal file warning in/home/junji-hashimoto/git/hasktorch/libtorch-ffi/libtorch-ffi.cabal@78:2: The field "cxx-options" is available since Cabal [2,1]
libtorch-ffi> test (suite: spec, args: -m Cuda)


Cuda
  CudaSpec
    When CUDA is out of memory, do GC and retry FAILED [1]

Failures:

  test/CudaSpec.hs:26:5:
  1) Cuda.CudaSpec When CUDA is out of memory, do GC and retry
       uncaught exception: CppException
       CppStdException "Exception: [enforce fail at CPUAllocator.cpp:64] . DefaultCPUAllocator: can't allocate memory: you tried to allocate 80000000 byte
s. Error code 12 (Cannot allocate memory)\nframe #0: c10::ThrowEnforceNotMet(char const*, int, char const*, std::string const&, void const*) + 0x47 (0x7f14c5b36e17 in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libc10.so)\nframe #1: c10::alloc_cpu(unsigned long) + 0x1ec (0x7f14c5b1dbfc in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libc10.so)\nframe #2: <unknown function> + 0x194fa (0x7f14c5b1f4fa in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libc10.so)\nframe #3: at::native::empty_cpu(c10::ArrayRef<long>, c10::TensorOptions const&, c10::optional<c10::MemoryFormat>) + 0xd3 (0x7f14b2585c03 in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libtorch.so)\nframe #4: <unknown function> + 0xe8f948 (0x7f14b273b948 in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libtorch.so)\nframe #5: <unknown function> + 0xcda493 (0x7f14b2586493 in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libtorch.so)\nframe #6: at::native::zeros(c10::ArrayRef<long>, c10::TensorOptions const&) + 0x29 (0x7f14b258c8a9 in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libtorch.so)\nframe #7: at::TypeDefault::zeros(c10::ArrayRef<long>, c10::TensorOptions const&) + 0x53 (0x7f14b28df393 in /home/junji-hashimoto/git/hasktorch/deps/libtorch/lib/libtorch.so)\nframe #8: /home/junji-hashimoto/git/hasktorch/libtorch-ffi/.stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/spec/spec() [0x808bf8]\nframe #9: /home/junji-hashimoto/git/hasktorch/libtorch-ffi/.stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/spec/spec() [0x66a841]\n; type: c10::Error"

  To rerun use: --match "/Cuda/CudaSpec/When CUDA is out of memory, do GC and retry/"

Randomized with seed 1105989586

Finished in 0.0026 seconds
1 example, 1 failure

libtorch-ffi> Test suite spec failed
Test suite failure for package libtorch-ffi-1.1.0.0
    spec:  exited with: ExitFailure 1
Logs printed to console

```

**Pull request merged**